scripts/helpers/check_zap_against_xml.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2023 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import os
+import sys
+import xml.etree.ElementTree as ET
+
+import click
+
+DATA_MODEL_PATH = 'data_model/clusters'
+
+
+@click.command()
+@click.option(
+    '--zap',
+    required=True,
+    type=str,
+    help='ZAP file')
+@click.option(
+    '--matter-root',
+    required=True,
+    type=str,
+    help='Path to the matter root')
+def main(zap, matter_root):
+
+    all_checks_successful = True
+
+    CLUSTER_REVISIONS_BY_CLUSTER_ID = {}
+
+    print('Getting latest cluster revisions from XML data model')
+    for file in os.listdir(os.path.join(matter_root, DATA_MODEL_PATH)):
+        if file.endswith(".xml"):
+            with open(os.path.join(matter_root, DATA_MODEL_PATH, file), "r") as cluster_xml_file:
+                print(f'Processing {file}')
+                xml_et = ET.parse(cluster_xml_file)
+                root = xml_et.getroot()
+
+                try:
+                    CLUSTER_REVISIONS_BY_CLUSTER_ID[int(root.attrib['id'], 16)] = root.attrib['revision']
```

Comment:
maybe we could add some heuristics here : if id starts with 0x then it is hex, otherwise decimal

Reply:
also `scripts/py_matter_idl/matter_idl/data_model_xml_parser.py` can parse these and will get the cluster ID correctly (except for those that do not have one, like base clusters), so you could use that too. Not sure if you want to complicate this PR more, so if you want to process things raw I guess  that works too.

Reply:
all the XMLs have IDs starting with 0x or the ID entry is still empty, which would need to get fixed once a cluster ID is assigned (EVSE-*.xml). or for some remaining clusters (e.g. OTASoftwareUpdate.xml) it might be an error in the scraper.

Reply:
Is that guaranteed? Why not just use ,0?

Reply:
Need to check the scrape tool in the DM-Editor. Waiting for access on that repo.

Reply:
@cecille The scraper seems to always output hex: https://github.com/csa-data-model/projects/blob/28aa06933b75edc9a8a3c2817df2ea62d5176fbb/DM-Editor/src/packages/dme-library/src/domain-model-utilities/cluster/services/xml-service/xml.builder.ts#L68